pkgs/top-level/all-packages.nix

```diff
@@ -29552,6 +29552,12 @@ with pkgs;
 
   grisbi = callPackage ../applications/office/grisbi { gtk = gtk3; };
 
+  gscreenshot = callPackage ../applications/graphics/gscreenshot {
+    inherit (python310Packages) buildPythonApplication pillow pygobject3
+    setuptools xlib;
+    inherit xclip xdg-utils;
+  };
```

Comment:
A few things: 

1. Why do you have `xclip ? null` and `xdg-utils ? null` in the argument list if you explicitly add them to the argument list anyway.
2. Do you actually need `python310` or any `python3` would do?
3. This is not the usual nixpkgs way of using `callPackage`. You should avoid using explicit arguments: I think you should instead do the following:
    * accept `python3Packages` directly, and use that to provide the packages to `propagatedBuildInputs`.
    * remove all explicit arguments from your `callPackage` call

Reply:
1. I had those default to `null` because they used to be not entirely essential when I first packaged it, though it seems that has since changed.
2. Just tried it, `python3` works just fine.
3. What I was initially thinking with this was to package an X11 version and a Wayland version, which would maybe require explicit `callPackage` arguments. Would this be appropriate? Would there be a more idiomatic way?

I'll push the changes for these, but I'd still very much like to ask for someone's input on how to best do this.

Reply:
For an X11 and Wayland version, I think you could do e.g: the default version is X11, and add a `gscreenshot-wayland` which overrides some of the arguments. However I don't really know the usual way this is handled in nixpkgs, I would defer to somebody more knowledgeable.